yt/frontends/amrvac/fields.py

```diff
@@ -29,14 +30,45 @@
     "spherical": ("r", "theta", "phi")
 }
 
+def _velocity(field, data, idir, prefix=None):
+    """Velocity = linear momentum / density"""
+    # This is meant to be used with functools.partial to produce
+    # functions with only 2 arguments (field, data)
+    # idir : int
+    #    the direction index (1, 2 or 3)
+    # prefix : str
+    #    used to generalize to dust fields
+    if prefix is None:
+        prefix = ""
+    moment = data["gas", "%smoment_%d" % (prefix, idir)]
+    rho = data["gas", "%sdensity" % prefix]
+
+    idx1 = np.where(rho == 0)[0]
+    if idx1.size > 0:
+        mylog.info("zeros found in %sdensity, patching them to compute corresponding velocity field." % prefix)
+        idx2 = np.where(moment == 0)[0]
+        if not np.all(np.in1d(idx1, idx2, assume_unique=True)):
+            raise RuntimeError
+        rho[idx1] = 1
+    return moment / rho
+
+code_density = "code_mass / code_length**3"
+code_moment = "code_mass / code_length**2 / code_time"
+code_pressure = "code_mass / code_length / code_time**2"
+
+# for now, define a finite family of dust fields (up to 100 species, should be enough)
+MAXN_DUST_SPECIES = 100
+known_dust_fields = [("rhod%d" % idust, (code_density, ["dust%d_density" % idust], None))
+                     for idust in range(1, MAXN_DUST_SPECIES+1)]
+for idir in range(1, 4):
```

Comment:
For this particular case, you could do `for idir in (1, 2, 3)` just for clarity, but not a big deal.

Reply:
I agree, this would be clearer.